noq_django/rest_api/api/caseworker_api.py

```diff
@@ -120,3 +122,62 @@ def set_booking_pending(request, booking_id: int):
         return booking
     except BookingStatus.DoesNotExist:
         raise HttpError(404, detail="Booking status does not exist.")
+
+@router.get("/guests/nights/count/{user_id}/{start_date}/{end_date}", response=UserShelterStayCountSchema, tags=["caseworker-frontpage"])
+def get_user_shelter_stay_count(request, user_id: int, start_date: str, end_date: str):
+    try:
+        start_date = date.fromisoformat(start_date)
+        end_date = date.fromisoformat(end_date)
+
+        user_bookings = Booking.objects.filter(
+            user_id=user_id,
+            start_date__lte=end_date,
+            end_date__gte=start_date
+        ).select_related(
+            'product__host__region'
+        ).prefetch_related(
+            'product__host__caseworkers'  
+        )
+
+        total_nights = 0
+        user_stay_counts = []
+        for booking in user_bookings:
+            nights = (min(booking.end_date, end_date) - max(booking.start_date, start_date)).days
+            if nights > 0:
+                total_nights += nights
+
+                host = booking.product.host  
+                host_data = {
+                    'id': host.id,
+                    'name': host.name,
+                    'street': host.street,
+                    'postcode': host.postcode,
+                    'city': host.city,
+                    'region': {
+                        'id': host.region.id,
+                        'name': host.region.name
+                    },
+                    'caseworkers': [cw.id for cw in host.caseworkers.all()]  
```

Comment:
I think it's better to exclude this in the HostSchema. I think with this change we expose information that should not be sent out to the client.

Reply:
Hur hoppar jag över caseworker? caseworker är ju inte definerad i HostSchema nu men HostSchema hämtar ju hela Host model och jag fattar det inte går att hoppa över att hämta caseworker. eller det gick ju inte innan för mig, eftersom jag fick felet jag skrev om i gruppen.

Reply:
caseworker is now optional in Schema, and None in the api response